### PR TITLE
Add signature validation support

### PR DESCRIPTION
**Pull request by zackliu**
Add signature validation and meanwhile I found some provided function or attribute are actually not working. So, some updates about resolving property and validation are moved.

### CODE REVIEW

File: src/SignalRServiceExtension/TriggerBindings/Resolver/SignalRRequestResolver.cs [hunk 1]
```diff
@@ -26,8 +28,23 @@ public bool ValidateContentType(HttpRequestMessage request)
...
+            if (!string.IsNullOrEmpty(accessToken) &&
+                request.Headers.TryGetValues(Constants.AsrsSignature, out var values))
+            {
+                var signatures = SignalRTriggerUtils.GetSignatureList(values.FirstOrDefault());
+                if (signatures == null)
+                {
+                    return false;
+                }
+                using (var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(accessToken)))
+                {
```
Comment (JialinXin) [inline src/SignalRServiceExtension/TriggerBindings/Resolver/SignalRRequestResolver.cs]: Add comments to aware this algorithm should be synced with RT.

**Comment by JialinXin** [comment]
>             var attribute = parameterInfo.GetCustomAttribute<SignalRTriggerAttribute>(false);

why here explicit assign false inherit but inside `GetResolvedConnectionString` doesn't?

---
Refers to: src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs:37 in a331253. [](commit_id = a3312533cd75c6cfebb7f8ad381fd1c129b339f8, deletion_comment = False)

File: src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs [hunk 2]
```diff
@@ -33,8 +39,68 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
...
+                }
+                resolvedConnectionString = _nameResolver.Resolve(attribute.Default);
+            }
+
+            if (string.IsNullOrEmpty(resolvedConnectionString))
+            {
+                resolvedConnectionString = _options.ConnectionString;
+            }
+
+            return resolvedConnectionString;
```
Comment (JialinXin) [inline src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs]: return string.IsNullOrEmpty(resolvedConnectionString) ? _options.ConnectionString : resolvedConnectionString;

Comment (wanlwanl) [reply src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs]: use the Utils.FirstOrDefault?

File: src/SignalRServiceExtension/TriggerBindings/Resolver/SignalRRequestResolver.cs [hunk 3]
```diff
@@ -24,10 +35,31 @@ public bool ValidateContentType(HttpRequestMessage request)
...
+
+            if (!string.IsNullOrEmpty(accessToken) &&
+                request.Headers.TryGetValues(Constants.AsrsSignature, out var values))
+            {
+                var signatures = SignalRTriggerUtils.GetSignatureList(values.FirstOrDefault());
+                if (signatures == null)
+                {
+                    return false;
+                }
+                using (var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(accessToken)))
```
Comment (vicancy) [inline src/SignalRServiceExtension/TriggerBindings/Resolver/SignalRRequestResolver.cs]: maybe move to the protocol project which will be shared by both side?

File: src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs (hunk 2 above)
Comment (zackliu) [reply src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs]: But there're only two parameters, I think it's simple and efficient enough to use `?:`

File: src/SignalRServiceExtension/TriggerBindings/Resolver/SignalRRequestResolver.cs (hunk 3 above)
Comment (zackliu) [reply src/SignalRServiceExtension/TriggerBindings/Resolver/SignalRRequestResolver.cs]: As it's a common HMAC algorithm, and there're existing implementation for each language. I don't put it in protocol for now.

**Pull request merged**